examples/sycl_example_02.cpp

```diff
@@ -0,0 +1,152 @@
+  KHRONOS STANDARDS. THE UNMODIFIED, NORMATIVE VERSIONS OF KHRONOS
+  SPECIFICATIONS AND HEADER INFORMATION ARE LOCATED AT
+     https://www.khronos.org/registry/
+
+  THE MATERIALS ARE PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+  EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+  MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+  IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+  CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+  TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+  MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
+*/
+
+#include <algorithm>
+#include <iostream>
+#include <vector>
+
+#include <experimental/algorithm>
+#include <sycl/execution_policy>
+
+using namespace std::experimental::parallel;
+
+sycl::sycl_execution_policy<> sycl_policy;
+
+/* This sample tests the updated multi-kernel bitonic sort implementation.
+ * We use a sycl buffer to perform all operations on
+ * the device.
+ * Note that for the moment the sycl variants of the algorithm
+ *   are on the sycl namespace and not in std::experimental.
+ */
+template <typename T> inline T init_num(T num, int max) {
+  if (std::is_integral<T>::value) {
+    return num;
+  } else {
+    return num / max;
+  }
+}
+
+template <typename T> struct typename_as_str {};
+
+template <> struct typename_as_str<unsigned char> {
+  static constexpr const char *name_ = "uchar";
+};
+
+template <> struct typename_as_str<char> {
+  static constexpr const char *name_ = "char";
+};
+
+template <> struct typename_as_str<unsigned short> {
+  static constexpr const char *name_ = "ushort";
+};
+
+template <> struct typename_as_str<short> {
+  static constexpr const char *name_ = "short";
+};
+
+template <> struct typename_as_str<unsigned int> {
+  static constexpr const char *name_ = "uint";
+};
+
+template <> struct typename_as_str<int> {
+  static constexpr const char *name_ = "int";
+};
+
+
+template <> struct typename_as_str<unsigned long> {
+  static constexpr const char *name_ = "ulong";
+};
+
+template <> struct typename_as_str<long> {
+  static constexpr const char *name_ = "long";
+};
+
+template <> struct typename_as_str<float> {
+  static constexpr const char *name_ = "float";
+};
+
+template <> struct typename_as_str<double> {
+  static constexpr const char *name_ = "double";
+};
+
+template <typename T>
+bool test(const int minInputSizeLog2 = 3, const int maxInputSizeLog2 = 5) {
+
+  bool sorted = true;
+
+  std::cout << __FUNCTION__ << "<" << typename_as_str<T>::name_ << ">"
+            << std::endl;
+
+  for (int i = minInputSizeLog2; i <= maxInputSizeLog2; ++i) {
+
+    std::vector<T> v;
+    v.resize(1 << i);
+
+    std::cout << "in : ";
+    for (int j = 0; j < v.size(); ++j) {
+      v[j] = init_num(static_cast<T>((v.size() - 1) - j), v.size());
+      std::cout << (v[j]) << (j == v.size() - 1 ? "" : ", ");
+    }
```

Comment:
Please replace this with a call to std::copy and use an ostream_iterator. I don't mind if you make this a function a la `print_range(std::cout, v)`.